include/votca/csg/topology.h

```diff
@@ -453,15 +455,13 @@ inline Molecule *Topology::CreateMolecule(std::string name) {
 }
 
 inline Residue *Topology::CreateResidue(std::string name, Index id) {
-  Residue *res = new Residue(id, name);
-  _residues.push_back(res);
-  return res;
+  _residues.emplace_back(new Residue(id, name));
+  return _residues.back().get();
 }
 
 inline Residue *Topology::CreateResidue(std::string name) {
-  Residue *res = new Residue(_residues.size(), name);
-  _residues.push_back(res);
-  return res;
+  _residues.emplace_back(new Residue(_residues.size(), name));
```

Comment:
we have the ownership problem here again.

Reply:
What do you mean? _residues is a vector of unique_ptrs it owns the residue?

Reply:
okay can you use `make_unqiue<Residue>(size,name)` then, there should not be any `new`in the code